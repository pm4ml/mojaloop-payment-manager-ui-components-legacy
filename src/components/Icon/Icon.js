import './Icon.scss';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

import * as utils from '../../utils/common';
import Tooltip from '../Tooltip';

const iconSizes = {
  xs: 10,
  s: 12,
  m: 14,
  l: 16,
};

class Icon extends PureComponent {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }

  onClick(e) {
    if (this.props.onClick) {
      this.props.onClick(e);
    }
  }
  render() {
    const {
      id,
      className,
      style,
      size,
      name,
      fill,
      stroke,
      spin,
      tooltip,
      tooltipKind,
      tooltipDelay,
      tooltipPosition,
    } = this.props;

    const svgStyle = {
      height: `${size}px`,
      width: `${size}px`,
      fill,
      stroke,
      ...style,
    };
    const componentClassName = utils.composeClassNames([
      'el-icon',
      spin && 'el-icon--spin',
      className,
    ]);
    const svg = (
      <svg id={id} style={svgStyle} onClick={this.onClick} className={componentClassName}>
        <use xlinkHref={`#${name}`} />
      </svg>
    );
    if (tooltip) {
      return (
        <Tooltip label={tooltip} position={tooltipPosition} delay={tooltipDelay} kind={tooltipKind}>
          {svg}
        </Tooltip>
      );
    }
    return svg;
  }
}

Icon.propTypes = {
  id: PropTypes.string,
  style: PropTypes.shape(),
  className: PropTypes.string,
  size: PropTypes.number,
  name: PropTypes.string,
  fill: PropTypes.string,
  stroke: PropTypes.string,
  spin: PropTypes.bool,
  onClick: PropTypes.func,
  tooltipKind: PropTypes.oneOf(['regular', 'error', 'info', 'warning', 'neutral']),
  tooltipDelay: PropTypes.number,
  tooltip: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
  tooltipPosition: PropTypes.oneOf(['top', 'bottom', 'left', 'right']),
};
Icon.defaultProps = {
  id: undefined,
  style: undefined,
  className: undefined,
  size: 20,
  name: undefined,
  fill: undefined,
  stroke: undefined,
  spin: false,
  onClick: undefined,
  tooltipKind: undefined,
  tooltipDelay: undefined,
  tooltip: undefined,
  tooltipPosition: undefined,
};

export default Icon;
export { iconSizes };
