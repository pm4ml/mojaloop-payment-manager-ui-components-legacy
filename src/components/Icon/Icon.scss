@import '../../assets/styles/vars/colors.scss';

.el-icon {
  display: flex;
  flex: 0 0 auto;
}

.el-icon--spin {
  animation: spinIcon 1s linear infinite;
}

@keyframes spinIcon {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}
