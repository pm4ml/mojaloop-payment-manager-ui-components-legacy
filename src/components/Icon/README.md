# Icon

Renders a SVG icon by using the xLinkHref reference. The SVG should already be imported in order to be used.

### Proptypes

|       Name      |       Type       |              Description               | Optional |                  Values                  |   Default   |
|-----------------|------------------|----------------------------------------|----------|------------------------------------------|-------------|
| classname       | `string`         | defines the HTML classname             | x        |                                          | `undefined` |
| id              | `string`         | defines the HTML id                    | x        |                                          | `undefined` |
| style           | `object`         | defines the HTML style                 | x        |                                          | `undefined` |
| name            | `string`         | defines the icon name _xLinkHref_      |          |                                          | `undefined` |
| fill            | `string`         | defines the fill color                 |          |                                          | `undefined` |
| stroke          | `string`         | defines the stroke color               |          |                                          | `undefined` |
| spin            | `bool`           | defines whether the icon should rotate |          |                                          | `false`     |
| size            | `number`         | defines the icon height and width      |          |                                          | `20`        |
| tooltip         | `node`, `string` | defines the tooltip component or label |          |                                          | `undefined` |
| tooltipDelay    | `number`         | defines the delay in ms                |          |                                          | `undefined` |
| tooltipKind     | `string`         | defines the tooltip kind               |          | _regular, error, info, warning, neutral_ | `undefined` |
| tooltipPosition | `string`         | defines the tooltip position           |          | _top, bottom, left, right_               | `undefined` |
