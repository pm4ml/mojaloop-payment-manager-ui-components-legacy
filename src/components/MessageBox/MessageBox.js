import './MessageBox.scss';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

import { composeClassNames } from '../../utils/common';
import { Icon } from '../index';

const splitLines = (prev, curr) => [...prev, ...curr.split(`\n`)];

const getMessageComponent = (message, index) => (
  <div key={index} className="el-message-box__message">
    {message}
  </div>
);

const getMessages = message => {
  const subMessages = typeof message === 'string' ? [message] : message;
  return subMessages.reduce(splitLines, []).map(getMessageComponent);
};

class MessageBox extends PureComponent {
  render() {
    const {
      id,
      kind,
      style,
      active,
      icon,
      fill,
      message,
      center,
      size,
      fontSize,
      className,
      children,
    } = this.props;

    if (!message && !children) {
      return null;
    }

    const higherSize = Math.max.apply(Math, [icon ? size : 0, fontSize, 20]);

    const messageBoxClassName = composeClassNames([
      'el',
      'el-message-box',
      `el-message-box--${kind}`,
      active ? `el-message-box--active` : `el-message-box--inactive`,
      center && 'el-message-box--centered',
      className,
    ]);

    const messagesClassName = composeClassNames([
      'el-message-box__messages',
      center && 'el-message-box__messages--centered',
    ]);

    const messageBoxStyle = {
      ...style,
      padding: `${higherSize / 2}px`,
      borderWidth: `${parseInt(higherSize / 10, 10)}px`,
    };

    const messagesStyle = {
      fontSize: `${fontSize}px`,
    };

    let iconComponent = null;
    if (icon) {
      iconComponent = (
        <div className="el-message-box__icon-box" style={{ marginRight: `${higherSize / 2}px` }}>
          <Icon className="el-message-box__icon" name={icon} size={size} fill={fill} />
        </div>
      );
    }

    return (
      <div id={id} className={messageBoxClassName} style={messageBoxStyle}>
        {iconComponent}
        <div className={messagesClassName} style={messagesStyle}>
          {children || getMessages(message)}
        </div>
      </div>
    );
  }
}

MessageBox.propTypes = {
  className: PropTypes.string,
  style: PropTypes.shape(),
  kind: PropTypes.oneOf([
    'default',
    'primary',
    'secondary',
    'tertiary',
    'success',
    'danger',
    'warning',
    'dark',
    'light',
  ]),
  size: PropTypes.number,
  fontSize: PropTypes.number,
  fill: PropTypes.string,
  icon: PropTypes.string,
  active: PropTypes.bool,
  center: PropTypes.bool,
  message: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.string), PropTypes.string]),
  // TODO: Add iconPosition: PropTypes.oneOf(['left', 'right']),
};
MessageBox.defaultProps = {
  className: undefined,
  style: undefined,
  kind: 'default',
  size: 20,
  fontSize: 13,
  fill: undefined,
  icon: undefined,
  message: '',
  center: false,
  active: false,

  // TODO: Add iconPosition: 'left',
};

export default MessageBox;
