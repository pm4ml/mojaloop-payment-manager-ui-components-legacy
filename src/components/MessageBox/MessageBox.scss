@import '../../assets/styles/vars/colors.scss';
@import '../../assets/styles/vars/shadow.scss';

.el-message-box {
  display: flex;
  max-width: 100%;
  border-style: solid;
  border-radius: 5px;
  align-items: center;
  justify-content: flex-start;
  flex-direction: row;
  overflow: hidden;
  animation: fadeInOverlayMessage 0.2s ease-out 1 forwards;
}

.el-message-box--centered {
  align-items: center;
  justify-content: center;
}

.el-message-box--default {
  fill: #333;
  color: #333;
  border-color: #333;
  background-color: #eee;
}
.el-message-box--primary {
  fill: $color__main-primary;
  color: $color__main-primary;
  border-color: $color__main-primary;
  background-color: $shadow-color--primary-2;
}
.el-message-box--secondary {
  fill: $color__main-secondary;
  color: $color__main-secondary;
  border-color: $color__main-secondary;
  background-color: $shadow-color--secondary-2;
}
.el-message-box--tertiary {
  fill: $color__main-tertiary;
  color: $color__main-tertiary;
  border-color: $color__main-tertiary;
  background-color: $shadow-color--tertiary-2;
}
.el-message-box--success {
  fill: $color__main-success;
  color: $color__main-success;
  border-color: $color__main-success;
  background-color: $shadow-color--success-2;
}
.el-message-box--danger {
  fill: $color__main-danger;
  color: $color__main-danger;
  border-color: $color__main-danger;
  background-color: $shadow-color--danger-2;
}
.el-message-box--warning {
  fill: $color__main-warning;
  color: $color__main-warning;
  border-color: $color__main-warning;
  background-color: $shadow-color--warning-2;
}
.el-message-box--dark {
  fill: #000;
  color: #000;
  border-color: #000;
  background-color: $shadow-color--dark-2;
}
.el-message-box--light {
  fill: $color__white;
  color: $color__white;
  border-color: $color__white;
  background-color: shadow-color-0light;
}

.el-message-box--inactive {
  border-color: transparent;
}

.el-message-box__icon-box {
  flex: 0 0 auto;
  display: flex;
  align-items: center;
  justify-content: center;
}
.el-message-box__icon {
}
.el-message-box__messages {
  flex: 1 1 auto;
}
.el-message-box__messages--centered {
  flex: 0 0 auto;
}

.el-message-box__message {
}

@keyframes fadeInOverlayMessage {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
