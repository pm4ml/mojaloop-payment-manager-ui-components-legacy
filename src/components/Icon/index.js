import Icon, { iconSizes } from './Icon';

export default Icon;
export { iconSizes };
