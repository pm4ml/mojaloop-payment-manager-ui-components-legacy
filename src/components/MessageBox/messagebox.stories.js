import React from 'react';

import Button from '../Button';
import MessageBox from './MessageBox';
import Row from '../Row';
import Column from '../Column';

export default {
  title: 'MessageBox',
  component: MessageBox,
};

export const Default = () => (
  <div>
    <div className="m5">
      <MessageBox
        message="A Default MessageBox, when not wrapped in flexbox, will take 100% width"
        kind="default"
      />
    </div>
  </div>
);

export const AllKindsRegular = () => (
  <div>
    <div className="m5">
      <MessageBox message="Default" kind="default" />
    </div>
    <div className="m5">
      <MessageBox message="Primary" kind="primary" />
    </div>
    <div className="m5">
      <MessageBox message="Secondary" kind="secondary" />
    </div>
    <div className="m5">
      <MessageBox message="Tertiary" kind="tertiary" />
    </div>
    <div className="m5">
      <MessageBox message="Success" kind="success" />
    </div>
    <div className="m5">
      <MessageBox message="Danger" kind="danger" />
    </div>
    <div className="m5">
      <MessageBox message="Warning" kind="warning" />
    </div>
    <div className="m5">
      <MessageBox message="Dark" kind="dark" />
    </div>
    <div className="m5">
      <MessageBox message="Light" kind="light" />
    </div>
  </div>
);

export const AllKindsWithIcon = () => (
  <div>
    <div className="m5">
      <MessageBox icon="deploy-small" message="Default" kind="default" />
    </div>
    <div className="m5">
      <MessageBox icon="deploy-small" message="Primary" kind="primary" />
    </div>
    <div className="m5">
      <MessageBox icon="deploy-small" message="Secondary" kind="secondary" />
    </div>
    <div className="m5">
      <MessageBox icon="deploy-small" message="Tertiary" kind="tertiary" />
    </div>
    <div className="m5">
      <MessageBox icon="deploy-small" message="Success" kind="success" />
    </div>
    <div className="m5">
      <MessageBox icon="deploy-small" message="Danger" kind="danger" />
    </div>
    <div className="m5">
      <MessageBox icon="deploy-small" message="Warning" kind="warning" />
    </div>
    <div className="m5">
      <MessageBox icon="deploy-small" message="Dark" kind="dark" />
    </div>
    <div className="m5">
      <MessageBox icon="deploy-small" message="Light" kind="light" />
    </div>
  </div>
);

export const AllKindsActiveIcon = () => (
  <div>
    <div className="m5">
      <MessageBox active icon="deploy-small" message="Default" kind="default" />
    </div>
    <div className="m5">
      <MessageBox active icon="deploy-small" message="Primary" kind="primary" />
    </div>
    <div className="m5">
      <MessageBox active icon="deploy-small" message="Secondary" kind="secondary" />
    </div>
    <div className="m5">
      <MessageBox active icon="deploy-small" message="Tertiary" kind="tertiary" />
    </div>
    <div className="m5">
      <MessageBox active icon="deploy-small" message="Success" kind="success" />
    </div>
    <div className="m5">
      <MessageBox active icon="deploy-small" message="Danger" kind="danger" />
    </div>
    <div className="m5">
      <MessageBox active icon="deploy-small" message="Warning" kind="warning" />
    </div>
    <div className="m5">
      <MessageBox active icon="deploy-small" message="Dark" kind="dark" />
    </div>
    <div className="m5">
      <MessageBox active icon="deploy-small" message="Light" kind="light" />
    </div>
  </div>
);

export const Centered = () => (
  <div>
    <div className="m5">
      <MessageBox center message="Font Size 20" size={20} fontSize={20} icon="deploy-small" />
    </div>
  </div>
);

export const MultipleMessages = () => (
  <div>
    <div className="m5">
      <MessageBox
        message={['first line', 'second line', 'third line']}
        size={20}
        fontSize={20}
        icon="deploy-small"
      />
    </div>
  </div>
);

export const Children = () => (
  <div>
    <div className="m5">
      <MessageBox>
        <Button label="I am a child" style={{ marginLeft: 10 }} />
      </MessageBox>
    </div>
  </div>
);

export const FillIcon = () => (
  <div>
    <div className="m5">
      <MessageBox message="Filling the icon" icon="deploy-small" fill="#f00" />
    </div>
  </div>
);

export const Sizes = () => (
  <div>
    <div className="m5">
      <MessageBox message="Font Size 10" size={20} fontSize={10} icon="deploy-small" />
    </div>
    <div className="m5">
      <MessageBox message="Font Size 20" size={20} fontSize={20} icon="deploy-small" />
    </div>
    <div className="m5">
      <MessageBox message="Font Size 30" size={20} fontSize={30} icon="deploy-small" />
    </div>
    <div className="m5">
      <MessageBox message="Icon Size 10" size={10} fontSize={20} icon="deploy-small" />
    </div>
    <div className="m5">
      <MessageBox message="Icon Size 20" size={20} fontSize={20} icon="deploy-small" />
    </div>
    <div className="m5">
      <MessageBox message="Icon Size 30" size={30} fontSize={20} icon="deploy-small" />
    </div>
    <div className="m5">
      <MessageBox message="Font and Icon Size 10" size={10} fontSize={10} icon="deploy-small" />
    </div>
    <div className="m5">
      <MessageBox message="Font and Icon Size 20" size={20} fontSize={20} icon="deploy-small" />
    </div>
    <div className="m5">
      <MessageBox message="Font and Icon Size 30" size={30} fontSize={30} icon="deploy-small" />
    </div>
  </div>
);

export const ActiveSizes = () => (
  <div>
    <div className="m5">
      <MessageBox active message="Font Size 10" size={20} fontSize={10} icon="deploy-small" />
    </div>
    <div className="m5">
      <MessageBox active message="Font Size 20" size={20} fontSize={20} icon="deploy-small" />
    </div>
    <div className="m5">
      <MessageBox active message="Font Size 30" size={20} fontSize={30} icon="deploy-small" />
    </div>
    <div className="m5">
      <MessageBox active message="Icon Size 10" size={10} fontSize={20} icon="deploy-small" />
    </div>
    <div className="m5">
      <MessageBox active message="Icon Size 20" size={20} fontSize={20} icon="deploy-small" />
    </div>
    <div className="m5">
      <MessageBox active message="Icon Size 30" size={30} fontSize={20} icon="deploy-small" />
    </div>
    <div className="m5">
      <MessageBox
        active
        message="Font and Icon Size 10"
        size={10}
        fontSize={10}
        icon="deploy-small"
      />
    </div>
    <div className="m5">
      <MessageBox
        active
        message="Font and Icon Size 20"
        size={20}
        fontSize={20}
        icon="deploy-small"
      />
    </div>
    <div className="m5">
      <MessageBox
        active
        message="Font and Icon Size 30"
        size={30}
        fontSize={30}
        icon="deploy-small"
      />
    </div>
  </div>
);

export const WrappedIntoEachOther = () => (
  <div>
    <div className="m5">
      <MessageBox icon="deploy-small" size={50}>
        <div className="m5">
          <MessageBox center message="Font Size 10" size={10} fontSize={10} icon="deploy-small" />
        </div>
        <div className="m5">
          <MessageBox center message="Font Size 20" size={20} fontSize={20} icon="deploy-small" />
        </div>
        <div className="m5">
          <MessageBox center message="Font Size 20" size={30} fontSize={30} icon="deploy-small" />
        </div>
      </MessageBox>
    </div>
  </div>
);
