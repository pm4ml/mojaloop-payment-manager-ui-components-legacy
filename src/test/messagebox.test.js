import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

import MessageBox from '../components/MessageBox';
import Icon from '../components/Icon';

// Shallow render, no sub-components ( still available children html)
it('renders the component', () => {
  const wrapper = shallow(<MessageBox message="test message" icon="deploy-small" />);
  expect(wrapper.find('div.el-message-box').exists()).toBe(true);
  expect(wrapper.find(Icon).exists()).toBe(true);
});

it('renders the messages', () => {
  const wrapper = shallow(<MessageBox message="test message" icon="deploy-small" />);
  expect(wrapper.find('.el-message-box__messages')).toHaveLength(1);
  expect(wrapper.find('.el-message-box__messages').text()).toBe('test message');
});

it('renders the active prop', () => {
  const wrapper = shallow(<MessageBox message="test message" icon="deploy-small" active />);
  expect(wrapper.find('div.el-message-box').hasClass('el-message-box--active')).toBe(true);
});

it('renders the center prop', () => {
  const wrapper = shallow(<MessageBox message="test message" icon="deploy-small" center />);
  expect(wrapper.find('div.el-message-box').hasClass('el-message-box--centered')).toBe(true);
});

it('renders the icon', () => {
  const wrapper = shallow(<MessageBox message="test message" icon="deploy" />);
  expect(wrapper.find('.el-message-box__icon')).toHaveLength(1);
  expect(wrapper.find(Icon)).toHaveLength(1);
});

it('renders the icon with override fill prop', () => {
  const wrapper = shallow(<MessageBox message="test message" icon="deploy" fill="#000" />);
  expect(wrapper.find(Icon).prop('fill')).toBe('#000');
});

it('applies the prop fill', () => {
  const wrapper = shallow(<MessageBox message="test message" icon="deploy" fill="#fff" />);
  expect(wrapper.find(Icon).prop('fill')).toBe('#fff');
});

it('applies the prop className', () => {
  const wrapper = shallow(<MessageBox message="test message" icon="deploy" className="test" />);
  expect(wrapper.find('.test')).toHaveLength(1);
});

it('renders the prop id', () => {
  const wrapper = shallow(
    <MessageBox message="test message" icon="deploy" id="testMessageBoxId" />,
  );
  expect(wrapper.find('#testMessageBoxId')).toHaveLength(1);
  expect(wrapper.find('div.el-message-box').prop('id')).toBe('testMessageBoxId');
});

it('renders the default correct kind', () => {
  const wrapper = shallow(<MessageBox message="test message" icon="deploy" />);
  expect(wrapper.find('div.el-message-box').hasClass('el-message-box--default')).toBe(true);
});

it('renders all the kinds', () => {
  ['default', 'primary', 'secondary', 'tertiary', 'danger', 'warning', 'dark', 'light'].forEach(
    kind => {
      const wrapper = shallow(<MessageBox message="test message" icon="deploy" kind={kind} />);
      expect(wrapper.find('div.el-message-box').hasClass(`el-message-box--${kind}`)).toBe(true);
    },
  );
});

// Mount render, include sub-components
it('renders the inner icon component', () => {
  const wrapper = mount(<MessageBox message="test message" icon="deploy" />);
  expect(wrapper.find('.el-icon')).toHaveLength(1);
});

// Snapshot testing
it('renders the button correctly when multiple props are set', () => {
  const wrapper = shallow(
    <MessageBox
      message="test message"
      icon="deploy-small"
      kind="secondary"
      disabled
      onClick={jest.fn}
    />,
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
