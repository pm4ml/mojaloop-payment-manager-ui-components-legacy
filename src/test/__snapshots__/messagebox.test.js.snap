// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the button correctly when multiple props are set 1`] = `
<div
  className="el el-message-box el-message-box--secondary el-message-box--inactive"
  style={
    Object {
      "borderWidth": "2px",
      "padding": "10px",
    }
  }
>
  <div
    className="el-message-box__icon-box"
    style={
      Object {
        "marginRight": "10px",
      }
    }
  >
    <Icon
      className="el-message-box__icon"
      name="deploy-small"
      size={20}
      spin={false}
    />
  </div>
  <div
    className="el-message-box__messages"
    style={
      Object {
        "fontSize": "13px",
      }
    }
  >
    <div
      className="el-message-box__message"
      key="0"
    >
      test message
    </div>
  </div>
</div>
`;
